src/libclipper/src/redis.cpp

```diff
@@ -157,46 +155,46 @@ std::vector<VersionedModelId> str_to_models(const std::string& model_str) {
   auto split =
       start + model_str.substr(start, end - start).find(ITEM_PART_CONCATENATOR);
   std::string model_name = model_str.substr(start, split - start);
-  std::string model_version_str = model_str.substr(split + 1, end - split - 1);
-  int version = std::stoi(model_version_str);
-  models.push_back(std::make_pair(model_name, version));
+  std::string model_version = model_str.substr(split + 1, end - split - 1);
+  models.push_back(std::make_pair(model_name, model_version));
 
   return models;
 }
 
 bool set_current_model_version(redox::Redox& redis,
-                               const std::string& model_name, int version) {
+                               const std::string& model_name,
+                               const std::string& version) {
   if (send_cmd_no_reply<string>(
           redis, {"SELECT", std::to_string(REDIS_METADATA_DB_NUM)})) {
     std::string key = gen_model_current_version_key(model_name);
-    const vector<string> cmd_vec{"SET", key, std::to_string(version)};
+    const vector<string> cmd_vec{"SET", key, version};
 
     return send_cmd_no_reply<string>(redis, cmd_vec);
   } else {
     return false;
   }
 }
 
-int get_current_model_version(redox::Redox& redis,
-                              const std::string& model_name) {
+std::string get_current_model_version(redox::Redox& redis,
```

Comment:
@nishadsingh1 Please return an [`optional`](http://www.boost.org/doc/libs/1_64_0/libs/optional/doc/html/index.html) here instead of an empty string.